File: 214. Shortest Palindrome/Python/Shortest Palindrome.py
class Solution(object):
    def shortestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        reverse = s[::-1]
        for i in range(len(s) + 1):
            if s.startswith(reverse[i:]):
                return reverse[:i] + s

File: 214. Shortest Palindrome/C++/Shortest Palindrome.cpp
class Solution {
public:
    string shortestPalindrome(string s) {
        string rev(s);
        reverse(rev.rbegin(), rev.rend());
        for (int i = 0; i < s.length(); i++) {
            if (s.find(rev.substr(i)) == 0) {
                return rev.substr(0, i) + s;
            }
        }
        return "";
    }
};